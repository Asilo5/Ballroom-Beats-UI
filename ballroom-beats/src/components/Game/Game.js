import React, { Component } from 'react';
import DanceFloor from '../DanceFloor/DanceFloor'
import { 
  View, 
  Text, 
  StyleSheet, 
  Animated, 
  TouchableOpacity 
} from 'react-native';
import { createStackNavigator } from 'react-navigation-stack';
import { withNavigation } from 'react-navigation';
import { Audio } from 'expo-av';
import EStyleSheet from 'react-native-extended-stylesheet';

class Game extends Component {
  constructor(props) {
    super(props)
    this.state= {
      renderDance: true
    }
    this.danceTime = null
  };

  async componentDidMount() {
    // let timeToEnd = this.props.navigation.getParam('song', '') * 1000
    let timeToEnd = 75000 // changed for LANDSCAPE VIEW STYLING ******************
    this.danceTime = setTimeout(() => this.setState({renderDance: false}), timeToEnd);
    let songPath = this.props.navigation.getParam('song', '').url
    this.backgroundMusic = new Audio.Sound();
    try {
      await this.backgroundMusic.loadAsync(
        this.checkSongSwitch(songPath)
      );
      await this.backgroundMusic.playAsync();
      await this.backgroundMusic.setIsLoopingAsync(false);
    } catch (error) {
       console.log("The music isn't playing")
    }
  };

  checkSongSwitch = (path) => {
    switch (path) {
      case "beyond-the-sea":
        return require('../../../assets/Music/Beyond_-_the_-_Sea.mp3');
      case "deja-vu":
        return require('../../../assets/Music/Deja_-_vu.mp3');
      case "game-of-thrones":
        return require('../../../assets/Music/Game_-_of_-_Thrones.mp3');
      case "melancolia-tropical":
        return require('../../../assets/Music/Melancolia_-_Tropical.mp3');
      default:
        return require('../../../assets/Music/Never_-_Gonna_-_Give_-_You_-_Up.mp3');
    }
  };

  _start = (sequenceTiming) => {
    Animated.loop(
      Animated.sequence(sequenceTiming), {iterations: -1, useNativeDriver: true}).start();
  };

  _onQuitPress = () => {
    this.backgroundMusic.stopAsync();
    this.props.navigation.navigate('End');
  };

  stopMusic = (time) => {
    setTimeout(() => {
      this.backgroundMusic.stopAsync();
      // const scoreText = this.getScore(getCounterValue, getExpectedValue)
      // this.props.navigation.navigate('End', {scoreText: scoreText});
    }, time);
  };

  stopDance = (getCounterValue, getExpectedValue) => {
    this.props.navigation.navigate('End', {scoreText: this.getScore(getCounterValue, getExpectedValue)});
  };

  getScore = (getCounterValue, getExpectedFloat, moose) => {
    let getExpectedValue = Math.floor(getExpectedFloat)

    if (getCounterValue > getExpectedValue) {
      return `${getCounterValue} out of ${getExpectedValue} steps - stop stumbling around!`
    } else if (getCounterValue < getExpectedValue) {
      return `${getCounterValue} out of ${getExpectedValue} steps - try to keep up!`
    } else {
      return `${getCounterValue} out of ${getExpectedValue} steps - you're perfect!  You don't need this app!`
    }
  };

  render() {
      return (
        <>
          <View style={styles.container}>
          {this.state.renderDance ? <DanceFloor start={this._start} song={this.props.navigation.getParam('song', '')} dance={this.props.navigation.getParam('dance', '')} stopMusic={this.stopMusic} stopDance={this.stopDance}/> : null}
            <View style={styles.btnContainer}>
                <TouchableOpacity
                    style={styles.button}
                    onPress={this._onQuitPress}>
                    <Text style={styles.quit}>Quit</Text>
                </TouchableOpacity>
             </View>
          </View>
        </>
      )
  }
};

const AppNavigator = createStackNavigator({
    Game: {
      screen: Game,
    },
});

const styles = EStyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'black',
        alignItems: 'center',
        justifyContent: 'center',
    },
    btnContainer: {
        // backgroundColor: "black",
    },
    '@media (min-width: 500)': {
      btnContainer: {
        backgroundColor: '',
      }
    },
    button: {
        backgroundColor: '#A93133',
        margin: 10,
        width: '80%',
        borderRadius: 50
    },
    '@media (min-width: 500)': {
      button: {
        width: '30%',
      }
    },
    quit: {
        fontSize: 30,
        fontWeight: "bold",
        textAlign: "center",
        padding: 20
    },
    text: {
      color: 'white',
      fontSize: 20,
      display: 'flex',
      flexDirection: 'row',
    }
});

export default withNavigation(Game);
