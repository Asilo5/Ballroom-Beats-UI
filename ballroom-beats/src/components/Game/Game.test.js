import React from 'react';
import renderer from 'react-test-renderer';
import Game from '../Game/Game';

describe('Game', () => {

  const mockProps = {
     navigation: {
       navigate: jest.fn(),
       getParam: jest.fn().mockImplementation((paramType) => {
         if (paramType === "song") {
           return {
                    title: "Beyond the Sea",
                    spotifyid: "none",
                    url: "beyond-the-sea",
                    delay: 0.167,
                    avbarduration: 1,
                    duration: 172.47,
                    tempo: 136.482,
                    TimeSignature: 4
                  }
         }

         if (paramType === "dance") {
           return "Waltz"
         }
       }),
     }
   };

   it('should match snapshot', () => {
    const game = renderer.create(<Game navigation={mockProps.navigation} song={mockProps.navigation.getParam} dance={mockProps.navigation.getParam} />).toJSON();

    expect(game).toMatchSnapshot();
   });
});
