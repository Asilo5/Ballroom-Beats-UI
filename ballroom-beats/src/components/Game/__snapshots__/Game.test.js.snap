// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "95%",
        "width": null,
      }
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "95%",
            "width": null,
          },
          undefined,
        ]
      }
    />
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flex": 2,
          "height": "100%",
          "width": null,
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": null,
            },
            undefined,
          ]
        }
      />
      <View>
        <View
          style={
            Object {
              "height": 300,
              "left": 40,
              "marginBottom": 60,
              "marginTop": 20,
              "padding": 20,
            }
          }
        >
          <View
            style={
              Object {
                "display": "flex",
                "flexDirection": "column",
                "height": "60%",
                "width": 400,
              }
            }
          >
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "height": 100,
                  "justifyContent": "space-between",
                  "marginBottom": 100,
                  "width": "80%",
                }
              }
            >
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#F60091",
                      "borderRadius": 24,
                      "borderWidth": 24,
                      "margin": 20,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 20,
                      "left": 24,
                    }
                  }
                >
                  Left
                </Text>
                <View />
              </View>
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "borderColor": "#FFEB00",
                        "borderRadius": 24,
                        "borderWidth": 24,
                        "margin": 20,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                      }
                    }
                  />
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 20,
                        "left": 24,
                      }
                    }
                  >
                    Left
                  </Text>
                  <View />
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "borderColor": "#F6811F",
                        "borderRadius": 24,
                        "borderWidth": 24,
                        "margin": 20,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                      }
                    }
                  />
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 20,
                        "left": 24,
                      }
                    }
                  >
                    Right
                  </Text>
                  <View />
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "width": "80%",
                }
              }
            >
              <View
                style={
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "borderColor": "#03ABF0",
                        "borderRadius": 24,
                        "borderWidth": 24,
                        "margin": 20,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                      }
                    }
                  />
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 20,
                        "left": 24,
                      }
                    }
                  >
                    Left
                  </Text>
                  <View />
                </View>
                <View
                  accessible={true}
                  focusable={true}
                  isTVSelectable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "borderColor": "#6F2C8F",
                        "borderRadius": 24,
                        "borderWidth": 24,
                        "margin": 20,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                      }
                    }
                  />
                  <Text
                    style={
                      Object {
                        "color": "white",
                        "fontSize": 20,
                        "left": 24,
                      }
                    }
                  >
                    Right
                  </Text>
                  <View />
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                isTVSelectable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "borderColor": "#71C043",
                      "borderRadius": 24,
                      "borderWidth": 24,
                      "margin": 20,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                    }
                  }
                />
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 20,
                      "left": 24,
                    }
                  }
                >
                  Right
                </Text>
                <View />
              </View>
            </View>
          </View>
        </View>
        <View />
        <Text
          style={
            Object {
              "color": "#A9C344",
              "fontSize": 23,
              "marginTop": 15,
              "textAlign": "center",
              "textShadowColor": "#000",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 2,
            }
          }
        >
          Your Points: 
          0
        </Text>
        <Text
          style={
            Object {
              "color": "#A9C344",
              "fontSize": 23,
              "marginBottom": 10,
              "marginTop": 5,
              "textAlign": "center",
              "textShadowColor": "#000",
              "textShadowOffset": Object {
                "height": 2,
                "width": 2,
              },
              "textShadowRadius": 2,
            }
          }
        >
          Possible Points: 
          392
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "black",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#A93133",
          "borderRadius": 50,
          "margin": 10,
          "opacity": 1,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 20,
            "textAlign": "center",
          }
        }
      >
        Quit
      </Text>
    </View>
  </View>
</View>
`;
